src/slot_machine.c

```diff
@@ -2178,83 +2378,105 @@ static bool8 AreTagsAtPosition_Reel1(s16 pos, u8 tag1, u8 tag2)
     return FALSE;
 }
 
-static bool8 AreCherriesOnScreen_Reel1(s16 offsetFromCenter)
+// Returns true if there are cherries on screen in reel 1 after the given number
+// of turns.
+static bool8 AreCherriesOnScreen_Reel1(s16 turns)
 {
-    if (GetTag(LEFT_REEL, 1 - offsetFromCenter) == GFXTAG_CHERRY
-     || GetTag(LEFT_REEL, 2 - offsetFromCenter) == GFXTAG_CHERRY
-     || GetTag(LEFT_REEL, 3 - offsetFromCenter) == GFXTAG_CHERRY)
+    if (GetTag(LEFT_REEL, 1 - turns) == GFXTAG_CHERRY ||
+        GetTag(LEFT_REEL, 2 - turns) == GFXTAG_CHERRY ||
+        GetTag(LEFT_REEL, 3 - turns) == GFXTAG_CHERRY)
         return TRUE;
     else
         return FALSE;
 }
 
-static bool8 IsBiasTowardsCherryOr7s(void)
+static bool8 BiasedTowardCherryOr7s(void)
 {
-    if (sSlotMachine->luckyFlags & (LUCKY_BIAS_777 | LUCKY_BIAS_MIXED_777 | LUCKY_BIAS_CHERRY))
+    if (sSlotMachine->machineBias & (BIAS_STRAIGHT_7 | BIAS_MIXED_7 | BIAS_CHERRY))
         return TRUE;
     else
         return FALSE;
 }
 
-static bool8 DecideReelTurns_BiasTag_Reel1_Bet1(u8 tag1, u8 tag2)
+// If a biased tag appears in the center of reel 1 within the next 4 turns, stop
+// there. That tag becomes the biasTag for the subsequent reels.
+static bool8 DecideStop_Bias_Reel1_Bet1(u8 tag1, u8 tag2)
 {
     s16 i;
 
-    for (i = 0; i < 5; i++)
+    for (i = 0; i <= 4; i++)
```

Comment:
Why `<=` here (and subsequently)?

Reply:
This is to emphasize the emphasize the maximum number of extra turns (4) the machine will add to manipulate the results. I added a constant to make this clearer.